giraffa-core/src/main/java/org/apache/giraffa/fileidrowkey/FileIdRowKeyFactory.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.giraffa.fileidrowkey;
+
+import org.apache.giraffa.RowKey;
+import org.apache.giraffa.RowKeyFactory;
+import org.apache.hadoop.hbase.client.Table;
+
+import java.io.IOException;
+
+public class FileIdRowKeyFactory extends RowKeyFactory {
+
+  private FileIdAgent agent;
+
+  @Override // RowKeyFactory
+  public void initialize(Table nsTable) {
+    agent = new FileIdAgent(nsTable);
+  }
+
+  @Override // RowKeyFactory
+  protected RowKey getRowKey(String src,
+                             long inodeId)
+      throws IOException {
+    return null; // TODO
```

Comment:
The full FileIdRowKey implementation is intended to be part of #34. I introduced a dummy implementation here as Konstantin asked me to demonstrate how the service architecture could be used in practice. I am not sure if we will be committing it here or in #34.